yjit/src/asm/x86_64/mod.rs

```diff
@@ -1034,6 +1025,25 @@ pub fn mov(cb: &mut CodeBlock, dst: X86Opnd, src: X86Opnd) {
     };
 }
 
+/// A variant of mov used for always writing the value in 64 bits for GC offsets.
+pub fn movabs(cb: &mut CodeBlock, dst: X86Opnd, value: u64) {
+    match dst {
+        X86Opnd::Reg(reg) => {
+            if reg.num_bits == 16 {
```

Comment:
Since `movabs` is for 64 bit only we should `assert_eq!(64, reg.num_bits)` and remove the branches below